lib/ansible/module_utils/iosxr.py

```diff
@@ -63,24 +98,270 @@ def check_args(module, warnings):
+
+    return module.connection
+
+
+def get_device_capabilities(module):
+    if hasattr(module, 'capabilities'):
+        return module.capabilities
+
+    capabilities = Connection(module._socket_path).get_capabilities()
+    module.capabilities = json.loads(capabilities)
+
+    return module.capabilities
+
+
+def transform_reply():
+    reply = '''<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+    <xsl:output method="xml" indent="no"/>
+
+    <xsl:template match="/|comment()|processing-instruction()">
+        <xsl:copy>
+            <xsl:apply-templates/>
+        </xsl:copy>
+    </xsl:template>
+
+    <xsl:template match="*">
+        <xsl:element name="{local-name()}">
+            <xsl:apply-templates select="@*|node()"/>
+        </xsl:element>
+    </xsl:template>
+
+    <xsl:template match="@*">
+        <xsl:attribute name="{local-name()}">
+            <xsl:value-of select="."/>
+        </xsl:attribute>
+    </xsl:template>
+    </xsl:stylesheet>
+    '''
+    if sys.version < '3':
+        return reply
+    else:
+        print("utf8")
+        return reply.encode('UTF-8')
+
+
+# Note: Workaround for ncclient 0.5.3
+def remove_namespaces(rpc_reply):
+    xslt = transform_reply()
+    parser = etree.XMLParser(remove_blank_text=True)
+    xslt_doc = etree.parse(BytesIO(xslt), parser)
+    transform = etree.XSLT(xslt_doc)
+
+    return etree.fromstring(str(transform(etree.parse(StringIO(str(rpc_reply))))))
+
+
+# Builds netconf xml rpc document from meta-data
+# e.g:
+#
+# Module inputs:
+# banner_params = [{'banner':'motd', 'text':'Ansible banner example', 'state':'present'}]
+#
+# Meta-data definition
+# bannermap = collections.OrderedDict()
+# bannermap.update([
+#     ('banner', {'xpath' : 'banners/banner', 'tag' : True, 'attrib' : "operation"}),
+#     ('a:banner', {'xpath' : 'banner/banner-name'}),
+#     ('a:text', {'xpath' : 'banner/banner-text', 'operation' : 'edit'})
+# ])
+#
+# Fields:
+#   key: exact match to the key expected in arg spec (prefixes --> a: values from arg_spec, m: values from meta-data)
+#   xpath: xpath of the element (based on YANG model)
+#   tag: True if no text on the element
+#   attrib: attribute to be embedded in the element (e.g. cx:operation="merge")
+#   operation: if edit --> includes the element in edit_config() query else ignore for get() queries
+#   value: if key is prefixed with "m:", provider value from here
+#   leaf: True --> if there is only one tag element under a subtree
+#
+# Output:
+# <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
+#   <banners xmlns="http://cisco.com/ns/yang/Cisco-IOS-XR-infra-infra-cfg">
+#     <banner xc:operation="merge">
+#       <banner-name>motd</banner-name>
+#       <banner-text>Ansible banner example</banner-text>
+#     </banner>
+#   </banners>
+# </config>
+
+def build_xml_subtree(container_ele, xmap, param=None, opcode=None):
```

Comment:
check `HAS_LXML` flag

Reply:
fixed: added in parent method: build_xml()